### PR TITLE
Allow covariance matrix for noise generation

### PR DESCRIPTION
**Pull request by yger**
We want to make the noise generation more realistic, and to do so, we can allow a covariance matrix in order to make the noise structure more realistic. However, I know that after discussion with @samuelgarcia , the goal is to keep the generation file as simple as possible, so maybe a more complicated object could be put elsewhere. Open to discussion

### CODE REVIEW

File: src/spikeinterface/core/generate.py [hunk 1]
```diff
@@ -2006,6 +2040,19 @@ def generate_ground_truth_recording(
...
+        cov_matrix = np.zeros((num_channels, num_channels), dtype=np.float32)
+        for i in range(num_channels):
+            for j in range(num_channels):
+                if i != j:
+                    cov_matrix[i, j] = (0.5 * correlated_noise) / distances[i, j]
+                else:
+                    cov_matrix[i, j] = 1
+
+    else:
+        cov_matrix = None
```
Comment (alejoe91) [inline src/spikeinterface/core/generate.py:2054]: Looks familar :P

**Review by alejoe91 — APPROVED** [review]

**Comment by yger** [comment]
Indeed, MEArec is back! After discussing with @DradeAW we realized the noise structure might be crucial for benchmarking. How did you calibrate the noise in mearec? We could discuss that more in-depth, because what would be the key components to make it more plausible. But then as Sam told me, the idea is to keep the generation as simple as possible...

**Comment by alejoe91** [comment]
My suggestion is the following:

- we keep a minimal generate.py in core, with very simple generation stuff (no ellipsoid, complex noises, etc)
- we move the complex machinery to the generation module and make ismy as complex as we want!

How does that sound?

**Comment by yger** [comment]
This sounds right, I'm open to anything. I'll check with @samuelgarcia how to organize such move.

**Comment by samuelgarcia** [comment]
Done in #2683.
So close.

**Pull request closed without merging**